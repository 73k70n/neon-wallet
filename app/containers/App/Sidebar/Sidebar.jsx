// @flow
import React, { Fragment } from 'react'
import classNames from 'classnames'
import { NavLink } from 'react-router-dom'

import Logout from './Logout'
import Tooltip from '../../../components/Tooltip'
import { ROUTES } from '../../../core/constants'

import HomeIcon from '../../../assets/navigation/home.svg'
import HistoryIcon from '../../../assets/navigation/history.svg'
import SendIcon from '../../../assets/navigation/send.svg'
import ReceiveIcon from '../../../assets/navigation/receive.svg'
import ContactsIcon from '../../../assets/navigation/contacts.svg'
import TokenSaleIcon from '../../../assets/navigation/tokens.svg'
import SettingsIcon from '../../../assets/navigation/settings.svg'
import NewsIcon from '../../../assets/navigation/news.svg'
import LightLogoWithoutText from '../../../assets/images/logo-without-text-black.png'
import DarkLogoWithoutText from '../../../assets/images/logo-without-text.png'

import styles from './Sidebar.scss'

type Props = {
  className: string,
  theme: ThemeType,
  pendingTransactionsCount: number,
  count: number,
  isWatchOnly?: boolean,
}

const LogoWithTooltipAndBlockHeight = ({
  count,
  theme,
}: {
  count: number,
  theme: string,
}) => {
  const themeBasedLogo =
    theme === 'Light' ? LightLogoWithoutText : DarkLogoWithoutText
  return (
    <Tooltip position="left">
      <div className={styles.logo} id="neon-logo-container">
        <img src={themeBasedLogo} id="neon-logo" alt="neon-logo" />
      </div>

      <div id="block-height-container" className={styles.blockHeight}>
        {count && (
          <Fragment>
            <div id="block-height-label" className={styles.heightText}>
              CURRENT BLOCK:
            </div>
            <div id="block-height">{count.toLocaleString()}</div>
          </Fragment>
        )}
      </div>
    </Tooltip>
  )
}

const Sidebar = ({
  className,
  theme,
  pendingTransactionsCount,
  isWatchOnly,
  count,
}: Props) => (
  <div className={classNames(styles.container, className)}>
    <div className={styles.group}>
      <LogoWithTooltipAndBlockHeight theme={theme} count={count} />
      <NavLink
        id="dashboard"
        exact
        to={ROUTES.DASHBOARD}
        className={styles.navItem}
        activeClassName={styles.active}
      >
        <HomeIcon />
        <div> Wallet </div>
      </NavLink>

      <NavLink
        id="history"
        exact
        to={ROUTES.TRANSACTION_HISTORY}
        className={styles.navItem}
        activeClassName={styles.active}
      >
        {pendingTransactionsCount > 0 && (
          <div className={styles.pendingTransactionsCount}>
            {pendingTransactionsCount}
          </div>
        )}
        <HistoryIcon />
        <div> Activity </div>
      </NavLink>

      <NavLink
        id="send"
        exact
        to={ROUTES.SEND}
        className={styles.navItem}
        activeClassName={styles.active}
      >
        <SendIcon />
        <div> Send </div>
      </NavLink>

      <NavLink
        id="receive"
        exact
        to={ROUTES.RECEIVE}
        className={styles.navItem}
        activeClassName={styles.active}
      >
        <ReceiveIcon />
        <div> Receive </div>
      </NavLink>

      <NavLink
        id="contacts"
        to={ROUTES.CONTACTS}
        className={styles.navItem}
        activeClassName={styles.active}
      >
        <ContactsIcon />
        <div> Contacts </div>
      </NavLink>

      {!isWatchOnly && (
        <NavLink
          id="tokensale"
          to={ROUTES.TOKEN_SALE}
          className={styles.navItem}
          activeClassName={styles.active}
        >
          <TokenSaleIcon />
          <div id="token-sale-label"> Token Sale </div>
        </NavLink>
      )}
      <NavLink
        id="News"
        to={ROUTES.NEWS}
        className={styles.navItem}
        activeClassName={styles.active}
      >
        <NewsIcon />
        <div> News </div>
      </NavLink>

      <NavLink
        id="settings"
        to={ROUTES.SETTINGS}
        className={styles.navItem}
        activeClassName={styles.active}
      >
        <SettingsIcon />
        <div> Settings </div>
      </NavLink>
    </div>

    <Logout
      className={classNames(
        styles.group,
        styles.logoutToolTipGroup,
        styles.navItem,
      )}
      id="logout"
    />
  </div>
)

export default Sidebar
