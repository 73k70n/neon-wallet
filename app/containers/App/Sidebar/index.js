// @flow
import { compose } from 'recompose'
import { withData } from 'spunky'
import { withRouter } from 'react-router-dom'

import Sidebar from './Sidebar'
import { addPendingTransaction } from '../../../actions/pendingTransactionActions'

const mapPendingTransactionsDataToProps = (
  pendingTransactions: Array<any>,
) => ({
  pendingTransactionsCount:
    (Array.isArray(pendingTransactions) && pendingTransactions.length) || 0,
})

export default compose(
  withRouter, // allow `NavLink` components to re-render when the window location changes
  withData(addPendingTransaction, mapPendingTransactionsDataToProps),
)(Sidebar)
