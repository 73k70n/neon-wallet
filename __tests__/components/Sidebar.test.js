import React from 'react'
import { shallow, mount } from 'enzyme'
import { createStore, provideStore, provideState } from '../testHelpers'

import Sidebar from '../../app/containers/App/Sidebar/Sidebar'

const initialState = {}
const store = createStore(initialState)
const wrapper = shallow(<Sidebar count={200} />)

describe('Sidebar', () => {
  test('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('renders block height when count prop present', () => {
    console.log({ wrapper: wrapper.debug() })
    const height = wrapper.find('#block-height')
    expect(height.text()).toEqual('200')
  })
})
