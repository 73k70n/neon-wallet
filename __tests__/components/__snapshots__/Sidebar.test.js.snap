// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders without crashing 1`] = `
<div
  className="container"
>
  <div
    className="group"
  >
    <div
      className="logo"
    >
      <img
        alt="neon-logo"
        src="test-file-stub"
      />
    </div>
    <div
      className="blockHeight"
    >
      <div
        className="heightText"
      >
        CURRENT BLOCK: 
      </div>
      <div
        id="block-height"
      >
        200
      </div>
    </div>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="navItem"
      exact={true}
      id="dashboard"
      to="/dashboard"
    >
      <Component />
      <div>
         Wallet 
      </div>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="navItem"
      exact={true}
      id="history"
      to="/transactions"
    >
      <Component />
      <div>
         Activity 
      </div>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="navItem"
      exact={true}
      id="send"
      to="/send/"
    >
      <Component />
      <div>
         Send 
      </div>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="navItem"
      exact={true}
      id="receive"
      to="/receive"
    >
      <Component />
      <div>
         Receive 
      </div>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="navItem"
      id="contacts"
      to="/contacts"
    >
      <Component />
      <div>
         Contacts 
      </div>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="navItem"
      id="tokensale"
      to="/token-sale"
    >
      <Component />
      <div>
         Token Sale 
      </div>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="navItem"
      id="News"
      to="/news"
    >
      <Component />
      <div>
         News 
      </div>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="navItem"
      id="settings"
      to="/settings"
    >
      <Component />
      <div>
         Settings 
      </div>
    </NavLink>
  </div>
  <Connect(withActions(Connect(withActions(Logout))))
    className="group logoutToolTipGroup navItem"
    id="logout"
  />
</div>
`;
